packages/flutter_tools/test/general.shard/android/android_device_test.dart

```diff
@@ -34,18 +34,36 @@ void main() {
   });
 
   group('getAdbDevices', () {
-    final MockProcessManager mockProcessManager = MockProcessManager();
+    MockProcessManager mockProcessManager;
+
+    setUp(() {
+      mockProcessManager = MockProcessManager();
+    });
+
     testUsingContext('throws on missing adb path', () {
       final Directory sdkDir = MockAndroidSdk.createSdkDirectory();
       Config.instance.setValue('android-sdk', sdkDir.path);
 
       final File adbExe = fs.file(getAdbPath(androidSdk));
       when(mockProcessManager.runSync(
         <String>[adbExe.path, 'devices', '-l'],
-      ))
-      .thenAnswer(
-        (_) => throw ArgumentError(adbExe.path),
-      );
+      )).thenThrow(ArgumentError(adbExe.path));
+      expect(() => getAdbDevices(), throwsToolExit(message: RegExp('Unable to find "adb".*${adbExe.path}')));
+    }, overrides: <Type, Generator>{
+      AndroidSdk: () => MockAndroidSdk(),
+      FileSystem: () => MemoryFileSystem(),
+      ProcessManager: () => mockProcessManager,
+    });
+
+    testUsingContext('throws on failing adb', () {
+      final Directory sdkDir = MockAndroidSdk.createSdkDirectory();
+      Config.instance.setValue('android-sdk', sdkDir.path);
+
+      final File adbExe = fs.file(getAdbPath(androidSdk));
+      when(mockProcessManager.runSync(
+        <String>[adbExe.path, 'devices', '-l'],
+      )).thenThrow(ProcessException(adbExe.path, <String>['devices', '-l']));
+      print(adbExe.path);
```

Comment:
minor nit: in case the `print` wasn't intended. :)

Reply:
Nope, thanks!